### PR TITLE
important fixes

### PR DESCRIPTION
**Pull request by tdowds**
This

1. Fixes the memory leak in our components subscribing to DelegateStore
2. Removes white space when loading assignments
3. Adds a forgotten break statement in the dispatch method of DelegateStore
4. Removes unneeded Assignment code in AdvisorRosterView

### CODE REVIEW

File: huxley/www/js/components/AdvisorAssignmentsView.js [hunk 1]
```diff
@@ -81,6 +81,10 @@ var AdvisorAssignmentsView = React.createClass({
     });
   },
 
+  componentWillUnmount: function() {
+    this._delegatesToken.remove();
```
Comment (kmeht) [inline huxley/www/js/components/AdvisorAssignmentsView.js]: Guard against the possibility of the component never mounting with:
```js
this._delegatesToken && this._delegatesToken.remove();
```

File: huxley/www/js/components/AdvisorRosterView.js [hunk 2]
```diff
@@ -68,6 +56,10 @@ var AdvisorRosterView = React.createClass({
     });
   },
 
+  componentWillUnmount: function() {
+    this._delegatesToken.remove();
```
Comment (kmeht) [inline huxley/www/js/components/AdvisorRosterView.js]: ditto

File: huxley/www/js/stores/DelegateStore.js [hunk 3]
```diff
@@ -78,6 +78,7 @@ class DelegateStore extends Store {
         break;
       case ActionConstants.UPDATE_DELEGATES:
         this.updateDelegates(action.schoolID, action.delegates);
+        break;
```
Comment (kmeht) [inline huxley/www/js/stores/DelegateStore.js]: Didn't we have a test case that would've caught this?

Comment (tdowds) [inline huxley/www/js/stores/DelegateStore.js]: I think the test tests whether the store was updated, not whether the store was updated and a change emitted

**Pull request merged**